### PR TITLE
Implement `ampersand-app` for ampersand CLI

### PR DESCRIPTION
**Pull request by fyockm**
as mentioned in AmpersandJS/ampersand-app#4

### CODE REVIEW

File: template/shared/client/app.js [hunk 1]
```diff
@@ -13,41 +13,47 @@ var domReady = require('domready');
...
+
+            // init our URL handlers and the history tracker
+            router: new Router(),
+
+            // This is how you navigate around the app.
+            // this gets called by a global click handler that handles
+            // all the <a> tags in the app.
+            // it expects a url without a leading slash.
+            // for example: "costello/settings".
+            navigate: function(page) {
```
Comment (mikehedman) [inline template/shared/client/app.js]: I'd kind of hoped that this method would get "promoted" into ampersand-app.  It seems like pretty much every ampersand application would need it, so it's always seemed wrong for it to have to be copy/pasted into the main app file.  Now that there's a parent, it seems like a great place to put it.

Comment (HenrikJoreteg) [inline template/shared/client/app.js]: @mikehedman I think there's some more we can do here while still maintaining flexibility.

I'm going to tweak some of this a bit and also go add some tests and such. Thanks for doing the initial conversion @fyockm. More thoughts incoming...

Comment (fyockm) [inline template/shared/client/app.js]: @mikehedman you make some very good points on both sides. I personally don't think `ampersand-app` (at least in its current incarnation) is the correct place for this abstraction, but perhaps there is place for this "boilerplate" type code?

I'll be interested to see what @HenrikJoreteg and crew have to say on the topic.

**Comment by HenrikJoreteg** [comment]
Wanted to iterate on this a bit in a single merged branch, as created in #96 which includes all these commits. Thanks @fyockm!

**Comment by HenrikJoreteg** [comment]
Let's continue this discussion in #96. I just made a few tweaks that I was envisioning there, @fyockm and  @mikehedman.

**Pull request closed without merging**